nomad/rpc.go

```diff
@@ -98,22 +99,44 @@ func (r *rpcHandler) listen(ctx context.Context) {
 			if r.shutdown {
 				return
 			}
-
-			select {
-			case <-ctx.Done():
-				return
-			default:
-			}
-
-			r.logger.Error("failed to accept RPC conn", "error", err)
+			r.handleAcceptErr(err, ctx)
 			continue
 		}
+		// No error, reset loop delay
+		r.acceptLoopDelay = 0
 
 		go r.handleConn(ctx, conn, &RPCContext{Conn: conn})
 		metrics.IncrCounter([]string{"nomad", "rpc", "accept_conn"}, 1)
 	}
 }
 
+// Sleep to avoid spamming the log, with a maximum delay according to whether or not the error is temporary
```

Comment:
Go has an idiom of starting method comments with the name of the method. No big deal.

```suggestion
// handleAcceptErr sleeps to avoid spamming the log, with a maximum delay
// according to whether or not the error is temporary.
```